app/wikipedia.py

```diff
@@ -20,4 +20,47 @@ def checkArticlesExists(pathList):
 						article["title"] = normalised["from"]
 		for article in queryJson["query"]["pages"]:
 			articleInfo[article["title"]] = not "missing" in article
-	return articleInfo
\ No newline at end of file
+	return articleInfo
+
+def checkValidPath(pathList):
+	for index in range(len(pathList)):
+		if index == len(pathList) - 1:
+			return True
+		
+		links = getArticleLinks(pathList[index])
+
+		if pathList[index + 1] not in links:
+			return False
+
+def getArticleLinks(article):
+	links = []
+	session = requests.Session()
+	url = "https://en.wikipedia.org/w/api.php"
+	params = {
+		"action": "query",
+		"format": "json",
+		"titles": article,
+		"prop": "links", 
+		"pllimit": "max"
+	}
+	
+	while True:
```

Comment:
This is a very inefficient way to do this - as it might take many many api calls to cycle through everything.

Instead of doing this api call in a loop, you should use
https://www.mediawiki.org/w/api.php?action=help&modules=query%2Blinks
the prop "pltitles". It should mean you will only have to make a request once - which means making far fewer api calls

Please let me know if you need help with this - I can check into your branch and update this myself if needed